### ISSUE TITLE
ACP-Forum&Posts-Forum Management-Copy forum is not working properly

### ISSUE THREAD

**Issue by Eldenroot** [mybb/mybb#3959]
If you wanna copy forum (and create a new one based on the setting below the copy section) with "Copy Forum settings nad properties" new forum is not created even you can see the notice about successful creation of the new forum.

![image](https://user-images.githubusercontent.com/778890/80302363-eafea080-87a9-11ea-8b9b-4352b16f3fb6.png)

![image](https://user-images.githubusercontent.com/778890/80302379-fbaf1680-87a9-11ea-9722-b5124adb1782.png)

**Comment by Sama34** [comment]
It seems to work for me.

**Comment by Eldenroot** [comment]
Hmm, weird. It happens only sometimes... if youor somebody else cannot reproduce, feel free to close :)

**Comment by Ben-MyBB** [comment]
Certainly works fine for me. Tested on a new install of 1.8.22. @yuliu Please can you check your board before we close this?

**Comment by yuliu** [comment]
Yeah, I have probably encountered a similar issue, but I can't recall correctly if it was the forum not being created or the forum's being created with not settings/permissions copied. I will try to have a test later.

**Comment by Sama34** [comment]
> or the forum's being created with not settings/permissions copied. I will try to have a test later.

I didn't check this, I also found it weird that you stay in the same page after creating a forum. I think it makes sense to be redirected to the forum settings page or at least to the forum page. Normal usage (i.e: not fresh installs) won't have you creating multiple forums frequently as to stay in that page, but because of this itself I also think it isn't worth changing.

**Comment by yuliu** [comment]
Have done several tests:
- For an existing forum and its permissions partly customized/inherited:
  - Copy settings & permissions to a new forum: new forum created and settings copied and all permissions copied except one user group. _Not checked here why the only one group's permission was not copied, maybe the permission to be copied is identical to target's parent's permission for this user group._ **Note here,** though some permissions are inherited but maybe they're not handled correctly, continue reading.
  - Copy permissions to an existing forum: all OK.

- For an existing forum and _all its permissions inherited from its parent_ (no custom permissions itself but its parent has custom permissions):
  - Copy permissions to an existing forum or a new forum: permissions are not handled, i.e., resulting in permissions of target forum being inherited from the root forum specified during copying. _And this is what I've been through before._

---

So for me, there are two issues:
1. Permissions for each user group which are inherited from parent forum are not handled correctly, in other words not copied at all.
2. a minor issue: admin log entry for copy settings & permissions to **a new forum** lacks of the new forum's name. E.g. `Copied settings and group permissions for usergroups #4,5,7,6,818,2,3,819,809,811,810,8,808,816,813,815,820,821,814,817,812 from forum #2070 (Test forum) to forum #2255 ()`, note the last empty brackets.

**Comment by Eldenroot** [comment]
Wow, great feedback and bug report! I know that somwthing is wrong. I think this is quite a high priority issue since it is a core feature which is quite often used :(

**Comment by Ben-MyBB** [comment]
@yuliu This could possibly relate to one of the oldest issues we have, https://github.com/mybb/mybb/issues/329

**Comment by yuliu** [comment]
> @yuliu This could possibly relate to one of the oldest issues we have, #329

They're both involving inherited permissions, but should be different things I think:
- This one, to me now, is having issues of inherited permissions are not copied using the _**copy forum**_ function.
- The problem in #329 seems like the default checked checkbox `Use Group Default` for each user group permissions using the _**Add new forum**_ / _**Add child forum**_ functions confuses forum owners. I haven't yet read relevant code and don't know if the functions are doing correctly here, but suggestions by @JN-Jones (https://github.com/mybb/mybb/issues/329#issuecomment-258899528) and further @WildcardSearch (https://github.com/mybb/mybb/issues/329#issuecomment-353484787) look good to me.

**Comment by arcalys** [comment]
Incorrectly referenced this one in my PR, although I did test the `forum copy` and inherited permissions were correctly copied.

**Comment by Ben-MyBB** [comment]
@arcalys Is this fixed?

**Comment by arcalys** [comment]
> [@arcalys](https://github.com/arcalys) Is this fixed?

It worked properly the last time I tested it

**Comment by Ben-MyBB** [comment]
Works fine here too.
